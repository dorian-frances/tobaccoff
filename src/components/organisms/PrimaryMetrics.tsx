import { StyleSheet, View } from 'react-native';
import PrimaryMetric, { PrimaryMetricProps } from '../molecules/PrimaryMetric';
import Divider from '../atoms/dividers/Divider';
import React from 'react';

export type PrimaryMetricsProps = {
  totalSavingsProps: PrimaryMetricProps;
  monthlySavingsProps: PrimaryMetricProps;
};

const PrimaryMetrics = ({
  totalSavingsProps,
  monthlySavingsProps,
}: PrimaryMetricsProps) => {
  return (
    <View style={styles.container}>
      <View style={styles.totalSavingsStyle}>
        <PrimaryMetric {...totalSavingsProps} />
      </View>
      <View style={styles.monthlySavingsStyle}>
        <PrimaryMetric {...monthlySavingsProps} />
      </View>
      <View style={styles.dividerStyle}>
        <Divider />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {},
  totalSavingsStyle: {},
  monthlySavingsStyle: {
    marginTop: 10,
  },
  dividerStyle: {
    marginTop: 30,
  },
});

export default PrimaryMetrics;
