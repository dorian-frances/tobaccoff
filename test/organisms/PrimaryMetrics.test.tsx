import { render } from '@testing-library/react-native';
import React from 'react';
import PrimaryMetrics from '../../src/components/organisms/PrimaryMetrics';

describe('<PrimaryMetrics/>', () => {
  it('should render correctly', () => {
    const component = render(
      <PrimaryMetrics
        totalSavingsProps={{
          sectionTextProps: {
            text: 'fakeTextPrimary',
            fontSize: 40,
          },
          metricTextProps: {
            metric: 20,
            unit: '€',
            minimumFractionDigits: 2,
            fontSize: 40,
          },
        }}
        monthlySavingsProps={{
          sectionTextProps: {
            text: 'fakeTextPrimary',
            fontSize: 40,
          },
          metricTextProps: {
            metric: 20,
            unit: '€',
            minimumFractionDigits: 2,
            fontSize: 40,
          },
        }}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
