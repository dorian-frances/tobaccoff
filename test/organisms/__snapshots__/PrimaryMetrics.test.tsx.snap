// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrimaryMetrics/> should render correctly 1`] = `
<View
  style={{}}
>
  <View
    style={{}}
  >
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontFamily": "Montserrat-Medium",
            "fontSize": 40,
          }
        }
      >
        fakeTextPrimary
      </Text>
      <Text
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            {
              "fontFamily": "Montserrat-Bold",
              "fontSize": 40,
            },
          ]
        }
      >
        20,00
        €
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontFamily": "Montserrat-Medium",
            "fontSize": 40,
          }
        }
      >
        fakeTextPrimary
      </Text>
      <Text
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            {
              "fontFamily": "Montserrat-Bold",
              "fontSize": 40,
            },
          ]
        }
      >
        20,00
        €
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "marginTop": 30,
      }
    }
  >
    <View
      style={
        {
          "borderColor": "#79747E",
          "borderWidth": 0.2,
        }
      }
    />
  </View>
</View>
`;
